lib/BuildSystem/BuildSystem.cpp

```diff
@@ -650,18 +650,18 @@ class DirectoryStructureInputNodeTask : public Task {
 /// This is the task to build a virtual node which isn't connected to any
 /// output.
 class VirtualInputNodeTask : public Task {
-  virtual void start(TaskInterface&) override {
+  virtual void start(TaskInterface) override {
   }
 
-  virtual void providePriorValue(TaskInterface&,
+  virtual void providePriorValue(TaskInterface,
                                  const ValueType& value) override {
   }
 
-  virtual void provideValue(TaskInterface&, uintptr_t inputID,
+  virtual void provideValue(TaskInterface, uintptr_t inputID,
                             const ValueType& value) override {
   }
 
-  virtual void inputsAvailable(TaskInterface& ti) override {
+  virtual void inputsAvailable(TaskInterface ti) override {
```

Comment:
Was this problem also an issue in C++? How come it never came up?

Reply:
Basically all of the internal  C++ uses of it where we actually hop threads used lambdas and were implicitly capturing copies.